app/controllers/users_controller.rb

```diff
@@ -79,7 +79,7 @@ def update
         if password_changed
           redirect_to new_user_session_path
         else
-          render :edit
+          redirect_to edit_user_path
```

Comment:
You are right, this breaks the validation errors. Will open a new issue for that instead of fixing it in this PR.